presto-spi/src/main/java/com/facebook/presto/spi/tracing/TracerProvider.java

```diff
@@ -16,9 +16,14 @@
 public interface TracerProvider
 {
     /**
-     * Initiate a trace.
      *
      * @return A @Tracer that should be kept throughout the whole duration of tracing.
      */
-    Tracer getNewTracer(boolean isTracingEnabled);
+    Tracer getNewTracer();
+
+    /**
+     *
+     * @return A @NoopTracer that does nothing.
+     */
+    Tracer getNoopTracer();
```

Comment:
We don't need this interface.... If we need to return a noop tracer, the callsite can direct inline the logic:
```java
if (isTracingEnabled) {
    return getNewTracer();
}
return NOOP_TRACER;